### DISCUSSION TITLE
If I want to develop and debug anaconda, How to get anaconda installing msg out pls?

### DISCUSSION THREAD

**Discussion by zhoupeng** [rhinstaller/anaconda#3403] [General]
Hi all, 

I wang to develop and debug anaconda and lorax.
I try to build an anaconda OS installer and then build an DVD ISO using mkisofs, with  fc34 repo. 
I test this DVD ISO throung installing it on an new virtuax box VM.

Then I get some error message during installing period, but I find it is not a goog way (I did'nt find) to get these log msg out of vm (A pic attatched  ![23](https://user-images.githubusercontent.com/1014132/119289752-bf383d80-bc7d-11eb-9722-7528cda2c805.png)
).

So I think using VM to test the installer may not easy to get err msg out (if there is, pls kindly tell me.). Is there any proper way (or developing environment) to develop and debug anaconda, that I can get the message easaly? Or is there some docs about runing anaconda on host to test the OS DVD/CD   installing directly, that the error msg is directly output to the host?

Thants all,
--
Best

**Comment by VladimirSlavik** [comment]
Hello zhoupeng!

Anaconda logs to several files under `/tmp`. Some of these are also shown in the tmux panes in tty1 - you get anaconda's stdout, shell, and tail for three of these logs (anaconda, storage, payload). Additionally, all logging is collected in syslog and journal. Something also goes to a virtio port `org.fedoraproject.anaconda.log.0` although I am not sure what exactly.

VMs are the standard method for the maintainer/developer team, Fedora QE, and Red Hat QE too. We use KVM with virt-manager. These tools probably integrate better with the host system because it is also Linux. I don't know how good is VirtualBox support; the virtio port is probably available only under KVM. I think this is all best summed up as "when working on Anaconda, you really want a Fedora host system and a guest VM".

Feel free to ask about more details!

**Reply by zhoupeng** [reply]
Hi Vladimír,

Thanks for your very helpful kindly reply.

By the way, I encounter an err  "some package has incorrect checksum" (attached as a picture at the end), while testing the installing of DVD ISO. 

I tried two ways to create  the rpm repository (shipped in the DVD's root dir, used to as repository source), which all got the same err  "some package has incorrect checksum" :
`
createrepo -d -s sha1 -g ${out_dir}/iso/Everything/repodata/comps.xml ${out_dir}/iso/Everything     // use sha1
`
`
mv ${out_dir}/iso/Everything/*  ${out_dir}/iso/     // Copy dir of Packages and repodata to the root of CD.
`

`
createrepo -d -g ${out_dir}/iso/Everything/repodata/comps.xml ${out_dir}/iso/Everything   // -s option was not set, to use default (that should be sha256)
`

I think it is the tools in Anaconda installer's rootfs do the checksum verification, so
I checked the installer's rootfs shipping anaconda (I think the rootfs should be in iso/images/install.img, if my understanding is wrong, pls correct me.):
`
$ sudo chroot ~/installer_rootfs/
`
`
bash-5.1# python3
`
`
python3    python3.9
`
`
bash-5.1# sha256sum --help
`
The version of python in the installer's rootfs is python3.9, and there is sha256sum tool too, So I think it shuold not be the problem like this [bug] (https://bobcares.com/blog/yum-error-performing-checksum/)

So I am confused when(where) Anaconda trigger the checksum verification of repository's packages. What configuration I missed, leads to this check err?

err snapshot:
![2](https://user-images.githubusercontent.com/1014132/119450220-2b8e6c00-bd66-11eb-83be-2fd1412022f8.png)

![3](https://user-images.githubusercontent.com/1014132/119449962-e0745900-bd65-11eb-89e7-7bce71a22593.png)

Thanks,
--
Best,

**Reply by VladimirSlavik** [reply]
I don't know much about this, but by chance, I found this article. It seems to be the same problem as you have. https://stackoverflow.com/questions/61449586/dnf-exceptions-error-some-packages-from-local-repository-have-incorrect-checksum

(We had some very weird crashes under VirtualBox, and could not find any cause. This could be it; I do not know yet because I just found the article.)

**Reply by zhoupeng** [reply]
Thank you very much. If I get the cause, I will response here. Many thanks.
